### PR TITLE
Use currency formatting for VPN display prices (Fixes #13476)

### LINKED ISSUE #13476: VPN Prices should be formatted for the locale of the page
We're currently getting this almost right, but in some cases the format of the number and symbol (e.g. `4.99 €` vs `€4,99`) vary quite a bit for different locales. As part of this we should move to letting Fluent handle this kind of formatting. 

`FluentNumber` is a [class in the Fluent system](https://projectfluent.org/python-fluent/fluent.runtime/stable/usage.html#numbers) that can format number input for a particular locale and currency. It would be nice if we could build in handling for currency formatting into the `ftl()` helper in bedrock for use in templates.

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

Uses babel's `format_currency` for displaying VPN subscription prices

## Significant changes and points to review

I've added a good number of tests for different currency / locale combinations, but this could do with some careful testing / spot checking that everything makes sense.

## Issue / Bugzilla link

#13476

## Testing

Demo: 

- https://www-demo1.allizom.org/en-US/products/vpn/
- https://www-demo1.allizom.org/an/products/vpn/pricing/

Local:

- http://localhost:8000/en-US/products/vpn/
- http://localhost:8000/en-US/products/vpn/pricing/

### CODE REVIEW

File: bedrock/products/templatetags/misc.py [hunk 1]
```diff
@@ -164,7 +164,10 @@ def vpn_monthly_price(ctx, plan=VPN_12_MONTH_PLAN, country_code=None, lang=None,
 
     available_plans = _vpn_get_available_plans(country_code, lang, bundle_relay)
     selected_plan = available_plans.get(plan, VPN_12_MONTH_PLAN)
-    amount = selected_plan.get("price")
+    price = float(selected_plan.get("price"))
+    currency = selected_plan.get("currency")
+    currency_locale = lang.replace("-", "_")
+    amount = format_currency(price, currency, locale=currency_locale)
```
Comment (alexgibson) [inline bedrock/products/templatetags/misc.py:170]: Note to self: need to figure out how to handle unknown locales here that throw an error:

e.g. `unknown locale 'ach'`

Comment (alexgibson) [reply bedrock/products/templatetags/misc.py:170]: Note that this is also an issue for Relay e.g. https://www-dev.allizom.org/ach/products/relay/ which is 500

Comment (robhudson) [reply bedrock/products/templatetags/misc.py:170]: Isn't `price` here something like `"US$4.99"`? In which case, `float` is going to throw an error.

I'm also curious if you need to convert currency strings formatted with different delimiters (e.g. `"4,99"`?

[`babel.numbers.parse_decimal`](https://babel.pocoo.org/en/latest/api/numbers.html#babel.numbers.parse_decimal) seems helpful. Plus, having a decimal vs a float is good for currencies b/c floats can end up like `4.9999988889` or something.

Looks like `format_currency` can also just take a string, so maybe we can avoid all of that anyway and pass it directly without converting it first?

Comment (alexgibson) [reply bedrock/products/templatetags/misc.py]: What I have here seems to work OK, and from what I can tell `format_currency` seems to use a sensible number of default decimals

**Comment by alexgibson** [comment]
This is ready for review but should not be merged until after all-hands

**Comment by alexgibson** [comment]
This is ready for review

**Review by craigcook — APPROVED** [review]
💱

**Pull request merged**